extensions/dicom-seg/src/loadSegmentation.js

```diff
@@ -0,0 +1,116 @@
+import OHIF from '@ohif/core';
+import * as dcmjs from 'dcmjs';
+import cornerstone from 'cornerstone-core';
+import cornerstoneTools from 'cornerstone-tools';
+
+const { DicomLoaderService } = OHIF.utils;
+
+export default async function loadSegmentation(
+  segDisplaySet,
+  referencedDisplaySet,
+  studies
+) {
+  const { studyInstanceUid } = referencedDisplaySet;
+
+  // Set here is loading is asynchronous.
+  // If this function throws its set back to false.
+  segDisplaySet.isLoaded = true;
+
+  const segArrayBuffer = await DicomLoaderService.findDicomDataPromise(
+    segDisplaySet,
+    studies
+  );
+
+  const dicomData = dcmjs.data.DicomMessage.readFile(segArrayBuffer);
+  const dataset = dcmjs.data.DicomMetaDictionary.naturalizeDataset(
+    dicomData.dict
+  );
+
+  dataset._meta = dcmjs.data.DicomMetaDictionary.namifyDataset(dicomData.meta);
+
+  const imageIds = _getImageIdsForDisplaySet(
+    studies,
+    studyInstanceUid,
+    referencedDisplaySet.seriesInstanceUid
+  );
+
+  const results = _parseSeg(segArrayBuffer, imageIds);
+
+  if (!results) {
+    throw new Error('Fractional segmentations are not yet supported');
+  }
+
+  const { labelmapBuffer, segMetadata, segmentsOnFrame } = results;
+  const { setters } = cornerstoneTools.getModule('segmentation');
+
+  // TODO: Could define a color LUT based on colors in the SEG.
+  // TODO -> Check the index and allocate a different labelmap if one exists?
```

Comment:
Remove this TODO, we do this the line below 👍 .

Reply:
Done, thanks.